### DISCUSSION TITLE
添加分贝检测组件

### DISCUSSION THREAD

**Discussion by HAHAHAHAHAYINING** [ClassIsland/ClassIsland#561] [插件功能请求]
### 检查清单

- [X] 我已将本应用更新到最新测试版 ![最新的版本号](https://img.shields.io/github/v/release/ClassIsland/ClassIsland?include_prereleases&style=flat-square&label=)，并看过[最新提交](https://github.com/ClassIsland/ClassIsland/commits/dev/)，确认我想要的功能还没有实现。
- [X] 我已在 [Issues](https://github.com/ClassIsland/ClassIsland/issues?q=label%3A新功能) 和 [ClassIsland 文档](https://docs.classisland.tech/zh-cn/latest/app/)中检索，确认我想要的功能未被提交过。
- [X] 我认为我想要的功能是用户广泛需要的，插件不能替代的，且没有超出 ClassIsland 作为**灵动课表**的开发目标，而非添加与课表及教学辅助无关的内容。（您可以学习[编写插件](https://docs.classisland.tech/zh-cn/latest/dev/plugins/create-project/)来实现特殊的需求）。
- [X] 我已知晓由于开发者时间精力有限，加上已经积压了大量的功能请求，我提交的功能请求可能需要比较久的时间才能实现。

### 背景与动机

课间或上课时，同学们经常讨论一些无关话题，影响老师授课或其他同学休息
![t018b32e8b5ad7454d9](https://github.com/user-attachments/assets/892dd3dd-61b1-414f-ace6-828b61dbd453)

### 想要实现或优化的功能

添加一个用于分贝检测的组件，可以在超出多少分贝时做出提醒

### 最后一步

- [X] 我认为上述的描述已经足以详细，以便开发人员理解您想要的功能。如果我的 issue 没有按照上述的要求填写，可能会被无条件关闭。

**Comment by DryIce-cc** [comment]
好活，插了（https://github.com/ClassIsland/ClassIsland/labels/%E6%8F%92%E4%BB%B6

**Comment by 0623jmq** [comment]
这个的话感觉上课时就得默认禁用了，因为有老师授课声音大且离电脑近等正常授课高音量情况。然后在托盘需要有临时开启分贝检测功能(“一节课，一个午，一天”按需选择)，以供上课但老师不讲课的情况。【建议仅供参考】

**Comment by liuziheng20091106** [comment]
检测到后除了提醒外，应当提供明显且易操作的弹窗来提供“本节课禁用”来避免有老师占自习课等尴尬情况（当然也要加“10分钟内不展示提示”来防止3秒弹窗一次的生草炸锅情况）

**Comment by BH0721** [comment]
> 检测到后除了提醒外，应当提供明显且易操作的弹窗来提供“本节课禁用”来避免有老师占自习课等尴尬情况（当然也要加“10分钟内不展示提示”来防止3秒弹窗一次的生草炸锅情况）

这种情况我设想的悬浮球似乎可以解决？

**Comment by xinya28** [comment]
与 #876 算同一个功能吗

**Reply by Peter267** [reply]
是的

**Reply by Yeson38** [reply]
但[#561](https://github.com/ClassIsland/ClassIsland/discussions/561)这个比[#876](https://github.com/ClassIsland/ClassIsland/discussions/876)要早诶

**Comment by Yeson38** [comment]
[Yeson38/Decibel_Monitor](https://github.com/Yeson38/Decibel_Monitor)做的很烂但也算有
